test/Microsoft.Framework.Internal.Test/BufferEntryCollectionTest.cs

```diff
@@ -0,0 +1,167 @@
+            Assert.Same(inner.BufferEntries, collection.BufferEntries[2]);
+        }
+
+        [Fact]
+        public void AddChar_ThrowsIfIndexIsOutOfBounds()
+        {
+            // Arrange
+            var collection = new BufferEntryCollection();
+
+            // Act and Assert
+            var ex = Assert.Throws<ArgumentOutOfRangeException>(
+                () => collection.Add(new[] { 'a', 'b', 'c' }, -1, 2));
+            Assert.Equal("index", ex.ParamName);
+            Assert.Equal(
+                "Specified argument was out of the range of valid values.\r\nParameter name: index",
+                ex.Message);
+        }
+
+        [Fact]
+        public void AddChar_ThrowsIfCountWouldCauseOutOfBoundReads()
+        {
+            // Arrange
+            var collection = new BufferEntryCollection();
+
+            // Act and Assert
+            var ex = Assert.Throws<ArgumentOutOfRangeException>(
+                () => collection.Add(new[] { 'a', 'b', 'c' }, 1, 3));
+            Assert.Equal("count", ex.ParamName);
+            Assert.Equal(
+                "Specified argument was out of the range of valid values.\r\nParameter name: count",
+                ex.Message);
+        }
+
+        public static TheoryData AddWithChar_RepresentsStringsAsChunkedEntriesData
+        {
+            get
+            {
+                return new TheoryData<char[], int, int, IList<object>> {
+                    {
+                        new[] { 'a' }, 0, 1, new[] { "a" }
+                    },
+                    {
+                        Enumerable.Repeat('a', 10).ToArray(),
+                        0,
+                        10,
+                        new[] { new string(Enumerable.Repeat('a', 10).ToArray()) }
+                    },
+                    {
+                        Enumerable.Repeat('b', 1024).ToArray(), 1, 1023, new[] { new string('b', 1023) }
+                    },
+                    {
+                        Enumerable.Repeat('c', 1027).ToArray(), 1, 1026, new[] { new string('c', 1024), "cc" }
+                    },
+                    {
+                        Enumerable.Repeat('d', 4099).ToArray(),
+                        2,
+                        4097,
+                        new[]
+                        {
+                            new string('d', 1024),
+                            new string('d', 1024),
+                            new string('d', 1024),
+                            new string('d', 1024),
+                            "d"
+                        }
+                    },
+                    {
+                        Enumerable.Repeat('e', 1025).ToArray(), 1023, 2, new[] { "ee" }
+                    }
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(AddWithChar_RepresentsStringsAsChunkedEntriesData))]
+        public void AddWithChar_RepresentsStringsAsChunkedEntries(char[] value, int index, int count, IList<object> expected)
```

Comment:
`AddWithChar_RepresentsStringsAsChunkedEntries` -> `AddChar_StoresArraysAsChunkedStrings` (stick with one name for that method and avoid the word "represents")